core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -5986,6 +5986,18 @@ private void checkLiteral2(String expression, String expected) {
         .ok(expected);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-4724">[CALCITE-4724]
+   * As for ClickHouse do not support values in from clause.
+   */
```

Comment:
1. the test descript should be the same as the CALCITE-4724.
2. the java doc lack of end label about "</a>"

Reply:
Updated . 
And add two more case in `testValues()` and `testValuesEmpty()`, please review again. tks